### ISSUE TITLE
[Bug] QR pairing broken

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/fenix#5990]
## Steps to reproduce
1. On Fenix, go to sync>Ready to scan
2. Scan the code from firefox.com/pair.
Case 1:
3. Approve from the desktop app.
4. Tap confirm on Fenix.
Case 2:
Confirm first on Fenix and then on desktop.
### Expected behavior
User logged in.
### Actual behavior
Case 1: Tapping confirm on Fenix doesn't do anything. Not logged in. 
The Cancel button works.

Case 2: Confirming first on Fenix then approving on desktop doesn't update and complete the flow on Fenix. It's blocked on waiting for approval.
### Device information

* Android device: Pixel 3 (Android 9), HTC Desire 820 (Android 6.0.1)
* Fenix version: Nightly 10/14

**Comment by sv-ohorvath** [comment]
RC 2.2.0-rc1 is crashing: 
[QR pairing logcat.txt](https://github.com/mozilla-mobile/fenix/files/3723786/QR.pairing.logcat.txt)

**Comment by pocmo** [comment]
```
10-14 12:28:16.815 10587-10657/? E/AndroidRuntime: FATAL EXCEPTION: pool-8-thread-1
    Process: org.mozilla.fenix, PID: 10587
    kotlin.KotlinNullPointerException
        at io.reactivex.plugins.RxJavaPlugins.throwNpe(RxJavaPlugins.java:1)
        at mozilla.components.service.fxa.manager.FxaAccountManager.postAuthenticated(FxaAccountManager.kt:41)
        at mozilla.components.service.fxa.manager.FxaAccountManager.stateActions(FxaAccountManager.kt:38)
        at mozilla.components.service.fxa.manager.FxaAccountManager$stateActions$1.invokeSuspend(FxaAccountManager.kt)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:2)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:15)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.j
```

**Comment by pocmo** [comment]
Adding [a-c] because of fxa in the stacktrace. But I'm not sure, needs verification.

**Comment by pocmo** [comment]
AC: https://github.com/mozilla-mobile/android-components/issues/4723

**Comment by grigoryk** [comment]
Seems like this is caused by the new sessionTokenScope; I'll admit that I didn't think to test pairing specifically when I landed that. For now, let's revert that patch, make sure it fixes the problem, and re-land it once underlying issue is fixed (see A-S bug: https://github.com/mozilla/application-services/issues/1979). We're not using this scope just yet, so it should be fine.

**Comment by softvision-miralobontiu** [comment]
This is not fixed yet.
I was able to reproduce the crash on Google Pixel (Android Q), and on Motorola Moto G6 (Android 8) the confirming page froze. Tested with v2.2.0-rc.2 build.

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest RC build v2.2.0-rc.2 (#12891716) with Sony Xperia Z2 (Android 6.0.1), Google Pixel (Android Q), Huawei Honor 8 (Android 7), and Google Pixel 3 (Android 9).

Not entirely fixed yet on the latest Nightly build 191017 (#12900613) with Google Pixel (Android Q) - I will file a new issue #6071.
